content/specs/xrpc.md

```diff
@@ -1,216 +1,137 @@
 
-Method schemas are encoded in JSON using [Lexicon Schema Documents](./lexicon).
+A common pattern in Lexicon design is to include a `cursor` parameter for pagination. The parameter should not include the parameter in the first request, and should keep all other parameters fixed between requests. If a cursor is included in a response, the next batch of responses can be fetched by including that value in a follow-on, continuing until the cursor is not included any longer, indicating the end of the result set has been reached.
 
-#### Schema distribution
+## Authentication
 
-Method schemas are designed to be machine-readable and network-accessible. While it is not currently _required_ that a schema is available on the network, it is strongly advised to publish schemas so that a single canonical & authoritative representation is available to consumers of the method.
+The current authentication scheme for XRPC is HTTP Bearer auth with JWT tokens, including refresh tokens. This scheme is likely to be extended or replaced.
 
-To fetch a schema, a request must be sent to the builtin [`getSchema`](#getschema) method. This request is sent to the authority of the NSID.
+Not all endpoints require authentication, but there is not yet a consistent way to enumerate which endpoints do or do not.
 
-### Built-in methods
+Initial login uses the `com.atproto.server.createSession` endpoint, including the password and an accoun tidentifier (eg, handle or registered email address). This returns a `refreshJwt` (as well as an initial `accessJwt`).
 
-#### `getSchema`
+Most requests should be authenticated using an access JWT, but the validity lifetime for these tokens is short. Every couple minutes, a new access JWT can be requested by hitting the `com.atproto.server.refreshSession` endpoint, using the refresh JWT instead of an access JWT.
 
-TODO
+The JWTs themselves should be treated as opaque tokens.
 
-### Requests
+### App Passwords
 
-#### HTTP Method
+App Passwords are a mechanism to reduce security risks when logging in to third-party clients and web applications. Accounts can create and revoke app passwords separate from their primary password. They are used to log in the same way as the primary password, but grant slightly restricted permissions to the client application, preventing destructive actions like account or changes to authentication settings (including app passwords themselves).
 
-The HTTP Method used depends on the `type` specified by the method schema.
+Clients and apps themselves do not need to do anything special to use app passwords. It is a best practice for most clients and apps to include a reminder to use an app password when logging in. App passwords usually have the form `xxxx-xxxx-xxxx-xxxx`, and clients can check against this format to prevent accidental logins with primary passwords (unless the primary password itself has this format).
 
-|Type|Method|
-|-|-|
-|`query`|`GET`|
-|`procedure`|`POST`|
+### Admin Token
```

Comment:
Interesting! I'm guessing this is primarily a legacy holdover, so it's more descriptive of the current prod PDS than normative, right? Otherwise it seems like this could be an implementation detail that doesn't really need to standardized, or if it does, maybe it would fit better in the protocol or repository docs?

Reply:
Yeah, all the auth stuff is sort of temporary and we expect to improve a lot on this. But that isn't going to be a quick fix, and we want to be transparent and clear how the existing scheme works in the meanwhile.

We do expect to have some of the other things, like repo history and sync stream, firmed up sooner than auth, which is why we just don't include those at all yet.